tests/FSharp.Compiler.ComponentTests/Conformance/UnionTypes/MultiCaseUnionStructTypes.fs

```diff
@@ -0,0 +1,250 @@
+            (Error 3204, Line 6, Col 12, Line 6, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 7, Col 12, Line 7, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+        ]
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 8`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of int * a: string
+    | B of string
+    | C of string
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3204, Line 5, Col 12, Line 5, Col 15, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 6, Col 12, Line 6, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 7, Col 12, Line 7, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+        ]
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 9`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of a: int * a1: string
+    | B of string
+    | C of string
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3204, Line 6, Col 12, Line 6, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 7, Col 12, Line 7, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+        ]
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 10`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
```

Comment:
This should keep working as it is now.
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEAzAzmgExAGoAfAOwEMBbGXAB0rBgAIAxCCAWACgBtADwBlAC5QArmBEA+ALq8RAT3qtREqQFVyASwjkWAXl4sTLUiwCCLCNhaUQLbeREsAVHYCMD3GKcBzY1NzACFrW1AWHyh/QJNzAGEwyN9yPyA==

Generated field a,a1,b,item.
Since there is a single "item", it works => keep it working for backwards compat.

Reply:
Updated